packages/account/src/Components/poi-country-selector/__tests__/poi-country-selector.spec.js

```diff
@@ -0,0 +1,106 @@
+import { isDesktop, isMobile } from '@deriv/shared';
+import CountrySelector from '../poi-country-selector';
+import { act } from 'react-dom/test-utils';
+import { Formik, Field } from 'formik';
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isDesktop: jest.fn(() => true),
+    isMobile: jest.fn(() => false),
+}));
+
+describe('<CountrySelector/>', () => {
+    let mock_props = {};
+
+    beforeEach(() => {
+        mock_props = {
+            handleSelectionNext: jest.fn(),
+            is_from_external: false,
+            residence_list: [
+                { value: 'Country 1', text: 'Country 1' },
+                { value: 'Country 2', text: 'Country 2' },
+                { value: 'Country 3', text: 'Country 3' },
+            ],
+            selected_country: '',
+            setSelectedCountry: jest.fn(),
+            show_helper_msg: '',
+        };
+    });
+
+    it('should render CountrySelector component not external with helper message', () => {
+        mock_props.show_helper_msg = true;
+
+        const { container } = render(<CountrySelector {...mock_props} />);
+
+        expect(screen.getByText('Proof of identity')).toBeInTheDocument();
+        expect(screen.getByText('In which country was your document issued?')).toBeInTheDocument();
+        expect(screen.getByText('Country')).toBeInTheDocument();
+        expect(screen.getByText('Try submitting an ID document instead.')).toBeInTheDocument();
+
+        const div_with_external_class = container.querySelector('.external-dropdown');
+        expect(div_with_external_class).not.toBeInTheDocument();
+    });
+
+    it('should render CountrySelector component external without hepler message', () => {
+        mock_props.is_from_external = true;
+
+        const { container } = render(<CountrySelector {...mock_props} />);
+
+        expect(screen.getByText('Proof of identity')).toBeInTheDocument();
+        expect(screen.getByText('In which country was your document issued?')).toBeInTheDocument();
+        expect(screen.getByText('Country')).toBeInTheDocument();
+        expect(screen.queryByText('Try submitting an ID document instead.')).not.toBeInTheDocument();
+
+        const div_with_external_class = container.querySelector('.external-dropdown');
+        expect(div_with_external_class).toBeInTheDocument();
+    });
+
+    it('should show error message after clicking the input without choosing the country', async () => {
+        render(<CountrySelector {...mock_props} />);
+
+        const field = screen.getByLabelText('Country');
+        const next_btn = screen.getByRole('button');
+
+        expect(field).toBeInTheDocument();
+        expect(screen.queryByText(/please select the country of document issuance/i)).not.toBeInTheDocument();
+        expect(screen.queryByText(/please select a valid country of document issuance/i)).not.toBeInTheDocument();
+
+        await waitFor(() => {
```

Comment:
I think we dont need to wrap `fireEvent` with await waitFor as  it causes some performance issues. could you please check this https://kentcdodds.com/blog/common-mistakes-with-react-testing-library

Reply:
Thanks for the article. Made a bookmark